diff --git a/src/screencast/ext_image_copy.c b/src/screencast/ext_image_copy.c
index afdfcd9..ec05fe1 100644
--- a/src/screencast/ext_image_copy.c
+++ b/src/screencast/ext_image_copy.c
@@ -18,7 +18,6 @@
 #include <wayland-client-protocol.h>
 #include <xf86drm.h>
 #include <sys/types.h>
-#include <pipewire/pipewire.h>
 
 #include "ext_image_copy.h"
 #include "screencast.h"
@@ -215,7 +214,6 @@ static void ext_frame_ready(void *data,
 		buffer->damage.size = 0;
 	}
 	cast->current_frame.damage.size = 0;
-	pw_stream_trigger_process(cast->stream);
 }
 
 static void ext_frame_failed(void *data,
diff --git a/src/screencast/pipewire_screencast.c b/src/screencast/pipewire_screencast.c
index 3c8f5c8..102a0f2 100644
--- a/src/screencast/pipewire_screencast.c
+++ b/src/screencast/pipewire_screencast.c
@@ -359,8 +359,6 @@ static void pwr_handle_stream_state_changed(void *data,
 	switch (state) {
 	case PW_STREAM_STATE_STREAMING:
 		cast->pwr_stream_state = true;
-		xdpw_pwr_dequeue_buffer(cast);
-		xdpw_wlr_frame_capture(cast);
 		break;
 	case PW_STREAM_STATE_PAUSED:
 		if (old == PW_STREAM_STATE_STREAMING) {
@@ -611,7 +609,6 @@ static void pwr_handle_stream_on_process(void *data) {
 	xdpw_wlr_frame_capture(cast);
 }
 
-
 static const struct pw_stream_events pwr_stream_events = {
 	PW_VERSION_STREAM_EVENTS,
 	.state_changed = pwr_handle_stream_state_changed,
@@ -655,8 +652,7 @@ void xdpw_pwr_stream_create(struct xdpw_screencast_instance *cast) {
 	pw_stream_connect(cast->stream,
 		PW_DIRECTION_OUTPUT,
 		PW_ID_ANY,
-		(PW_STREAM_FLAG_ALLOC_BUFFERS |
-			PW_STREAM_FLAG_DRIVER),
+		PW_STREAM_FLAG_ALLOC_BUFFERS,
 		params.data, params.size / sizeof(struct spa_pod *));
 
 	spa_pod_dynamic_builder_clean(&builder);
